c7n/actions/network.py

```diff
@@ -214,16 +254,31 @@ def get_groups(self, resources):
             isolation_groups = self.resolve_group_names(
                 r, self._get_array('isolation-group'), resolved_groups)
 
+            tag = self._get_array('add-by-tag')
+            if tag:
+                add_by_tag_groups = self.get_groups_by_tag(tag['key'],tag['values'],r['VpcId'])
```

Comment:
@naohito-intuit should we make this call once outside the loop rather than making the same call for each iteration of the resources enumeration loop?

Reply:
@darrendao That is a very good point. I moved the call outside the loop so that the API call happens only once when a user provides an **add-by-tag** attribute in action. It will search from list of JSON data response inside the loop.